examples/llm/common/content_extractor_module.py

```diff
@@ -0,0 +1,294 @@
+
+import logging
+import typing
+from concurrent.futures import ThreadPoolExecutor
+from dataclasses import dataclass
+from pathlib import Path
+
+import fsspec
+import mrc
+import mrc.core.operators as ops
+import pandas as pd
+from haystack import Document
+from haystack.nodes import DocxToTextConverter
+from haystack.nodes import PDFToTextConverter
+from haystack.nodes import TextConverter
+from haystack.nodes.file_converter import BaseConverter
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from pydantic import ValidationError
+
+from morpheus.messages import MessageMeta
+from morpheus.modules.schemas.examples.llm.content_extractor_schema import ContentExtractorSchema
+from morpheus.utils.module_utils import ModuleLoaderFactory
+from morpheus.utils.module_utils import register_module
+
+logger = logging.getLogger(__name__)
+
+ContentExtractorLoaderFactory = ModuleLoaderFactory("file_content_extractor",
+                                                    "morpheus_examples_llm",
+                                                    ContentExtractorSchema)
+
+
+@dataclass
+class FileMeta:
+    file_path: str
+    file_name: str
+    file_type: str
+
+
+class CsvTextConverter(BaseConverter):
+    """
+    Converts a CSV file column content to text documents.
+    """
+
+    outgoing_edges = 1
+
+    def convert(self,
+                file_path: Path | list[Path] | str | list[str] | list[Path | str],
+                meta: typing.Optional[dict[str, typing.Any]],
+                remove_numeric_tables: typing.Optional[bool] = None,
+                valid_languages: typing.Optional[list[str]] = None,
+                encoding: typing.Optional[str] = "UTF-8",
+                id_hash_keys: typing.Optional[list[str]] = None) -> list[Document]:
+        """
+        Load a CSV file and convert it to Documents.
+
+        Parameters
+        ----------
+        file_path:
+            Path to the CSV file you want to convert.
+        meta:
+            Optional dictionary of metadata key-value pairs that you want to append to the returned document.
+        encoding:
+            Optional file encoding format, default: `UTF-8`.
+        id_hash_keys:
+            Generates the document ID from a custom list of strings that refer to the document's attributes.
+        remove_numeric_tables: bool
+            Removes numeric tables from the csv.
+        valid_languages:
+            Valid languages
+
+        Returns
+        -------
+        list[haystack.Document]
+            List of documents, 1 document per line in the CSV.
+        """
+        if not isinstance(file_path, list):
+            file_path = [file_path]
+
+        docs: list[Document] = []
+        text_column_names = ["content"]
+
+        if meta is not None:
+            text_column_names = set(meta.get("csv", {}).get("text_column_names", text_column_names))
+
+        for path in file_path:
+            df = pd.read_csv(path, encoding=encoding)
```

Comment:
Any reason your not using `read_file_to_df`? If so maybe just add a comment

Reply:
Probably not, I'll update.